src/uct/base/uct_iface.c

```diff
@@ -773,30 +811,52 @@ ucs_status_t uct_ep_keepalive_create(pid_t pid, uct_keepalive_info_t **ka_p)
     return status;
 }
 
-ucs_status_t
-uct_ep_keepalive_check(uct_ep_h tl_ep, uct_keepalive_info_t **ka, pid_t pid,
-                       unsigned flags, uct_completion_t *comp)
+static ucs_status_t uct_iface_schedule_ep_err(uct_ep_h ep, ucs_status_t status)
+{
+    uct_base_iface_t *iface = ucs_derived_of(ep->iface, uct_base_iface_t);
+    uct_base_ep_error_handle_info_t *err_info;
+
+    if (iface->err_handler == NULL) {
+        ucs_diag("ep %p: unhandled error %s", ep, ucs_status_string(status));
+        return UCS_OK;
+    }
+
+    err_info = ucs_malloc(sizeof(*err_info), "uct_base_ep_err");
+    if (err_info == NULL) {
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    err_info->ep     = ep;
+    err_info->status = status;
+    ucs_callbackq_add_safe(&iface->worker->super.progress_q,
+                           uct_iface_ep_error_handle_progress, err_info,
+                           UCS_CALLBACKQ_FLAG_ONESHOT);
+    return UCS_OK;
+}
+
+ucs_status_t uct_ep_keepalive_check(uct_ep_h ep, uct_keepalive_info_t **ka,
+                                    pid_t pid, unsigned flags,
+                                    uct_completion_t *comp)
 {
     ucs_status_t status;
     ucs_time_t create_time;
 
     UCT_EP_KEEPALIVE_CHECK_PARAM(flags, comp);
 
-    if (ucs_unlikely(*ka == NULL)) {
+    if (*ka == NULL) {
```

Comment:
why `ucs_unlikely` was removed?

Reply:
since keepalive is nod considered fast path